### PR TITLE
feat: add isAuthCallback util and setup validations based on it - OKTA-285863

### PR DESCRIPTION
**Pull request by shuowu**
Changes: 

1. add `isOAuthCallback` function in oauthUtil module
2. guard token.get* (different getToken methods) with isOAuthCallback
3. guard tokeManager.get with isOAuthCallback (if autoRenew is on)
4. add unit tests
5. add error handling examples in README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1794,17 +1798,26 @@ By default, if no parameters are passed, both the access token and ID token obje
...
 ```javascript
-// In this example, the access token is stored under the key 'myAccessToken'
-const accessTokenObject = authClient.tokenManager.get('myAccessToken');
-// In this example, the ID token is stored under the key "myIdToken"
-const idTokenObject = authClient.tokenManager.get('myIdToken');
-authClient.token.getUserInfo(accessTokenObject, idTokenObject)
+// In this example, the access token is stored under the key 'myAccessToken', the ID token is stored under the key "myIdToken"
+Promise.all([
+  authClient.tokenManager.get('myAccessToken'),
+  authClient.tokenManager.get('myAccessToken')
```
Comment (pparlev) [inline README.md:1811]: Shouldn't this be `'myIdToken'` ?

Comment (shuowu) [reply README.md:1811]: Thanks for catching it 👍

**Comment by mabarbeau** [comment]
We use `?code=` in the url for non Okta 3rd party integrations. This change has caused us some serious issues.

**Comment by shuowu** [comment]
@mabarbeau Can you open a GitHub issue with detailed reproduction steps? We'll look into a proper solution for it.

**Comment by mabarbeau** [comment]
https://github.com/okta/okta-auth-js/issues/474

**Pull request merged**